src/cellars.rs

```diff
@@ -1,39 +1,120 @@
 use crate::{
     cork::cache::{self, is_approved},
     error::{Error, ErrorKind},
-    gas::CellarGas,
-    utils::get_eth_provider,
+    utils::sp_call_error,
 };
-use abscissa_core::tracing::{info, warn};
-use ethers::{abi::Address, prelude::*};
+use abscissa_core::tracing::info;
+use ethers::abi::Address;
 use std::result::Result;
 
 pub(crate) mod aave_v2_stablecoin;
+pub(crate) mod adaptors;
 pub(crate) mod cellar_v1;
 pub(crate) mod cellar_v2;
+pub(crate) mod cellar_v2_2;
 
-pub async fn get_gas_price() -> Result<U256, Error> {
-    if std::env::var("ETHERSCAN_API_KEY").is_ok() {
-        match CellarGas::etherscan_standard().await {
-            Ok(gas) => return Ok(gas),
-            Err(err) => {
-                warn!("failed to retrieve gas estimate from etherscan: {}", err);
-            }
-        }
+// constants
+// addresses are normalized by removing the 0x prefix and converting to lowercase for reliable comparison
+
+// block lists
+
+pub const BLOCKED_ADAPTORS: [&str; 3] = [
+    ADAPTOR_UNIV3_V1,
+    ADAPTOR_VESTING_SIMPLE_V1,
+    ADAPTOR_COMPOUND_C_TOKEN_V1,
+];
+pub const BLOCKED_POSITIONS: [u32; 9] = [4, 5, 6, 7, 8, 9, 10, 11, 12];
+
+// cellars
+
+pub const CELLAR_RY1INCH: &str = "c7b69e15d86c5c1581dacce3cacaf5b68cd6596f";
```

Comment:
This looks like we're missing the allowed catalogue/position updates we just added recently to 3.x-main for RYBTC/RYETH. We also still need the setup adaptor allowances for RYUSD as I do not believe they have been applied yet.

Reply:
Those are moved to governance now, so I removed them. Am I missing something?

Reply:
Yeah, I was thinking the wrong way about this. We may still want to add in references to the new cellars/adaptors though if we're going to keep these lists around, as most of the values are unused.